### PR TITLE
TD-2481 - Implement course progress view in refactored tracking system

### PR DESCRIPTION
**Pull request by Auldrin-Possa**
**JIRA link**
https://hee-tis.atlassian.net/browse/TD-2481

**Description**
Feature course progress view in refactored tracking system implemented.

**Screenshots**
image

image
image

-----
**Developer checks**

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

**Review by rshrirohit — CHANGES_REQUESTED** [review]
Please see these very small changes suggested in line.

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateProgressController.cs [hunk 1]
```diff
@@ -388,5 +398,82 @@ DelegateAccessRoute accessedVia
...
+            }
+            var model = new PreviewProgressViewModel(delegateCourseProgess, accessedVia);
+
+            var renderedViewHTML = RenderRazorViewToString(this, "DownloadProgress", model);
+
+            var delegateId = User.GetCandidateIdKnownNotNull();
+            var pdfReportResponse = await pdfService.PdfReport(progressId.ToString(), renderedViewHTML, delegateId);
+            if (pdfReportResponse != null)
+            {
+
```
[the remark this answers is not in the document]
Comment (Auldrin-Possa) [reply DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateProgressController.cs]: formatter was run on this file. If it is about new line, I have removed the same now.

File: DigitalLearningSolutions.Data/DataServices/ProgressDataService.cs [hunk 2]
```diff
@@ -348,6 +352,61 @@ FROM Progress
             ).SingleOrDefault();
         }
 
+        public DelegateCourseProgressInfo? GetDelegateCourseProgress(int progressId)
+        {
+            return connection.Query<DelegateCourseProgressInfo>(
+                @"SELECT p.ProgressID, Ce.CentreName, Ca.FirstName + ' ' + Ca.LastName AS CandidateName, A_1.ApplicationName + ' - ' + Cu.CustomisationName AS Course, p.Completed,
```
[the remark this answers is not in the document]
Comment (Auldrin-Possa) [reply DigitalLearningSolutions.Data/DataServices/ProgressDataService.cs]: Done. Please have a look.

File: DigitalLearningSolutions.Data/Models/Courses/DelegateCourseProgressInfo.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        public string CentreName { get; set; }
+        public string CandidateName { get; set; }
+        public string Course { get; set; }
+        public int DiagnosticAttempts { get; set; }
+        public int TotalTime { get; set; }
+        public int LearningDone { get; set; }
+        public int PLAttempts { get; set; }
+        public int PLPasses { get; set; }
+        public int Sections { get; set; }
+        public bool IsAssessed { get; set; }
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Data/Models/Courses/DelegateCourseProgressInfo.cs:17]: Please address this build warning as we are now moving towards addressing these as per the jira issue https://hee-tis.atlassian.net/browse/TD-2316

**Review by rshrirohit — CHANGES_REQUESTED** [review]

File: DigitalLearningSolutions.Data/Models/Courses/DelegateCourseProgressInfo.cs (hunk 3 above)
Comment (Auldrin-Possa) [reply DigitalLearningSolutions.Data/Models/Courses/DelegateCourseProgressInfo.cs]: Done.

**Pull request merged**